mypy/modulefinder.py

```diff
@@ -215,28 +215,31 @@ def _find_module(self, id: str) -> Optional[str]:
         near_misses = []  # Collect near misses for namespace mode (see below).
         for base_dir, verify in candidate_base_dirs:
             base_path = base_dir + seplast  # so e.g. '/usr/lib/python3.4/foo/bar/baz'
+            dir_prefix = base_dir
+            for _ in range(len(components) - 1):
```

Comment:
Might be better to strip off `dir_chain` directly with `base_dir[:len(dir_chain)]`?

Reply:
This doesn't work (also it looks like you forgot minus there). I also tried with plus or minus one and some tests fail.